plugins/shortcuts/src/ShortcutForm.tsx

```diff
@@ -50,6 +52,8 @@ export const ShortcutForm = ({
   allowExternalLinks,
 }: Props) => {
   const classes = useStyles();
+  const shortcutApi = useApi(shortcutsApiRef);
+  const shortcutData = shortcutApi.get();
```

Comment:
Hm. This performs the read during render which is an antipattern, and may lead to issues where an old shortcut snapshot is used for comparisons still even though they have changed since last time.

Note how the shortcuts display itself [handles the subscription](https://github.com/backstage/backstage/blob/cc14e1ba0ec638dbc9c07636e8db89e3cfd2e5cc/plugins/shortcuts/src/Shortcuts.tsx#L43) to the active state of shortcuts. Then you want to make your checker functions wrapped with `useCallback` with the list of shortcuts in their dependency arrays.

Reply:
sure added changes @freben